### PR TITLE
Feat: dev runtime

### PR DESCRIPTION
**Pull request by GopherJ**
1. [x] add dev & vanilla runtime
2. [x] make sure dev runtime works
3. [x] make sure vanilla runtime works
4. [x] squash commits

use `./target/release/parallel --dev` to run in dev mode (one node with aura consensus)

close: #38

### CODE REVIEW

**Comment by GopherJ** [comment]
@kaichaosun @yz89 @MrPai you can start reviewing

File: node/src/service.rs [hunk 1]
```diff
@@ -37,6 +48,332 @@ native_executor_instance!(
...
+    let keystore = keystore_container.sync_keystore();
+    SyncCryptoStore::insert_unknown(
+        &*keystore,
+        vanilla_runtime::pallet_ocw_oracle::KEY_TYPE,
+        secret_uri,
+        key_pair.public().as_ref(),
+    )
+    .expect("Insert key should succeed");
+
+    //todo change to mock pool
```
Comment (kaichaosun) [inline node/src/service.rs:133]: Is there issue to track this todo item?

File: node/Cargo.toml [hunk 2]
```diff
@@ -24,14 +24,17 @@ log                    = '0.4.13'
 primitives = { package = 'parallel-primitives', path = '../primitives' }
 
 pallet-transaction-payment-rpc = { version = '3.0.0', git = 'https://github.com/paritytech/substrate.git', branch = 'rococo-v1' }
-parachain-runtime              = { version = '2.0.0', path = '../runtime' }
+parachain-runtime              = { version = '2.0.0', path = '../runtime/parachain' }
```
Comment (kaichaosun) [inline node/Cargo.toml:27]: Probably update the version to 3.0

File: node/src/chain_spec.rs [hunk 3]
```diff
@@ -13,16 +13,20 @@
...
 use serde::{Deserialize, Serialize};
+use sp_consensus_aura::sr25519::AuthorityId as AuraId;
 use sp_core::{sr25519, Pair, Public};
+use sp_finality_grandpa::AuthorityId as GrandpaId;
 use sp_runtime::traits::{IdentifyAccount, Verify};
+use vanilla_runtime::{AuraConfig, GrandpaConfig};
 
 /// Specialized `ChainSpec` for the normal parachain runtime.
-pub type ChainSpec = sc_service::GenericChainSpec<parachain_runtime::GenesisConfig, Extensions>;
+pub type VanillaChainSpec =
```
Comment (kaichaosun) [inline node/src/chain_spec.rs:27]: Should be called ParachainSpec.

File: node/src/chain_spec.rs [hunk 4]
```diff
@@ -125,6 +139,100 @@ pub fn local_testnet_config(id: ParaId) -> ChainSpec {
...
+                .map(|k| (k, 1 << 60))
+                .collect(),
+        },
+        pallet_sudo: vanilla_runtime::SudoConfig { key: root_key },
+        orml_tokens: vanilla_runtime::TokensConfig {
+            endowed_accounts: endowed_accounts
+                .iter()
+                .flat_map(|x| {
+                    vec![
+                        (x.clone(), CurrencyId::DOT, 1_000 * TOKEN_DECIMAL),
```
Comment (kaichaosun) [inline node/src/chain_spec.rs:177]: As discussed, we should allow user to get specific token with a faucet.

File: node/src/chain_spec.rs [hunk 5]
```diff
@@ -125,6 +139,100 @@ pub fn local_testnet_config(id: ParaId) -> ChainSpec {
...
+            // total_supply: 1000 * TOKEN_DECIMAL, // 1000
+            // total_borrows: 600 * TOKEN_DECIMAL, // 600
+            borrow_index: RATE_DECIMAL,               // 1
+            exchange_rate: 2 * RATE_DECIMAL / 100,    // 0.02
+            base_rate: 2 * RATE_DECIMAL / 100,        // 0.02
+            multiplier_per_year: RATE_DECIMAL / 10,   // 0.1
+            jump_muiltiplier: 11 * RATE_DECIMAL / 10, // 1.1
+            kink: 8 * RATE_DECIMAL / 10,              // 0.8
+            collateral_rate: vec![
+                (CurrencyId::DOT, 5 * RATE_DECIMAL / 10),
```
Comment (kaichaosun) [inline node/src/chain_spec.rs:203]: Extract the common part to a const.

File: node/src/chain_spec.rs [hunk 6]
```diff
@@ -125,6 +139,100 @@ pub fn local_testnet_config(id: ParaId) -> ChainSpec {
...
+                (CurrencyId::xDOT, 5 * RATE_DECIMAL / 10),
+            ],
+            liquidation_incentive: vec![
+                (CurrencyId::DOT, 9 * RATE_DECIMAL / 10),
+                (CurrencyId::KSM, 9 * RATE_DECIMAL / 10),
+                (CurrencyId::BTC, 9 * RATE_DECIMAL / 10),
+                (CurrencyId::USDC, 9 * RATE_DECIMAL / 10),
+                (CurrencyId::xDOT, 9 * RATE_DECIMAL / 10),
+            ],
+            //FIXME :In fact,"liquidation_threshold" should be higher than "collateral_rate",
```
Comment (kaichaosun) [inline node/src/chain_spec.rs:216]: Can we have a issue to track it, and add more context in the issue?

**Review by kaichaosun — COMMENTED** [review]
When deploy to rococo, we needs to update dependencies in parallel to `rococo-v1` branch for substrate and cumulus, rococo now iterates fast, have this 2 runtime in same workspace may broke the vanilla runtime.

File: node/src/chain_spec.rs (hunk 6 above)
Comment (yz89) [reply node/src/chain_spec.rs:216]: Maybe issue #40 caused by the low liquidation_threshold.

File: node/src/chain_spec.rs (hunk 3 above)
Comment (GopherJ) [reply node/src/chain_spec.rs:27]: ParallelSpec

**Review by yz89 — APPROVED** [review]

**Review by MrPai — APPROVED** [review]

**Pull request merged**